pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java

```diff
@@ -155,6 +156,9 @@ public CompletableFuture<Boolean> canConsumeAsync(TopicName topicName, String ro
     @Override
     public CompletableFuture<Boolean> canLookupAsync(TopicName topicName, String role,
             AuthenticationDataSource authenticationData) {
+        if (SystemTopicNames.isTransactionCoordinatorAssign(topicName)) {
```

Comment:
You should write this code in the `org.apache.pulsar.broker.authentication.AuthenticationService`.

That looks like an important change, could you discuss this in the mailing list?

Reply:
Why should we implement it in `AuthenticationService`?